lib/modules/datasource/docker/common.ts

```diff
@@ -136,31 +136,33 @@ export async function getAuthHeaders(
       return opts.headers ?? null;
     }
 
-    let scope = `repository:${dockerRepository}:pull`;
+    const authUrl = new URL(`${authenticateHeader.params.realm}`);
+
     // repo isn't known to server yet, so causing wrong scope `repository:user/image:pull`
     if (
       is.string(authenticateHeader.params.scope) &&
       !apiCheckUrl.endsWith('/v2/')
     ) {
-      scope = authenticateHeader.params.scope;
+      authUrl.searchParams.append('scope', authenticateHeader.params.scope);
+    } else {
+      authUrl.searchParams.append(
+        'scope',
+        `repository:${dockerRepository}:pull`
+      );
```

Comment:
the else needs to be moved to the outer if, as it already checks for scope

Reply:
I am not following. Please suggest the code change.

Reply:
🤔 looks like I saw something wrong. sorry